File: src/OpenSBS/client/src/lib/angles.ts
export default class Angles {
    public static toDegrees(radians: number): number {
        return radians * (180 / Math.PI);
    }

    public static toRadians(degrees: number): number {
        return degrees * (Math.PI / 180);
    }

    public static bearingToString(bearing: number): string {
        return Math.round(bearing).toString().padStart(3, '0')
    }
}

File: src/OpenSBS.Core/Utils/Angles.cs
using System.Numerics;

namespace OpenSBS.Core.Utils
{
    internal static class Angles
    {
        public static double ToRadians(double degrees) =>
            degrees * (Math.PI / 180);

        public static double ToDegrees(double radians) =>
            radians * (180 / Math.PI);

        public static int ToBearing(Vector2 direction) =>
            (int)Math.Round((Math.Atan2(direction.Y, direction.X) + 360) % 360);
    }
}
